hammer/hammer.go

```diff
@@ -236,21 +236,22 @@ func (t *Throttle) Decrease() {
 
 func (t *Throttle) Run(ctx context.Context) {
 	ticker := time.NewTicker(1 * time.Second)
+Loop:
 	for {
 		select {
 		case <-ctx.Done(): //context cancelled
 			return
 		case <-ticker.C:
 			tokenCount := t.opsPerSecond
-			sessionOversupply := 0
+			timeout := time.After(1 * time.Second)
 			for i := 0; i < tokenCount; i++ {
 				select {
 				case t.tokenChan <- true:
-				default:
-					sessionOversupply += 1
+				case <-timeout:
+					t.oversupply = tokenCount - i
+					break Loop
```

Comment:
Won't this cause the throttle to entirely stop emitting tokens? Or is that what's required? (I sorta expected you wanted to break the `for ... tokenCount` loop, but maybe I've misunderstood)

Reply:
Good catch!

My excuse is that the monitor I use for coding has been turning itself off/switching inputs randomly so I had to race it to get this label in. Hopefully I've fixed it now :-)